tangelo/toolboxes/molecular_computation/rdms.py

```diff
@@ -41,6 +48,96 @@ def matricize_2rdm(two_rdm, n_orbitals):
     return rho
 
 
+def compute_rdms(ferm_ham, exp_data, mapping, up_then_down):
+    """
+    Computes the 1- and 2-RDM and their spin-summed versions
+    using a Molecule object and frequency data either in the form of a
+    classical shadow or a dictionary of frequency histograms.
+
+    Args:
+        ferm_ham (FermionicOperator): Fermionic operator with n_spinorbitals and n_electrons defined
+        exp_data (ClassicalShadow or dict): classical shadow or a dictionary of expectation values of qubit terms
+        mapping: qubit mapping
+        up_then_down: spin ordering for the mapping
+
+    Returns:
+        complex array: 1-RDM
+        complex array: 2-RDM
+        complex array: spin-summed 1-RDM
+        complex array: spin-summed 2-RDM
+    """
+    onerdm = np.zeros((ferm_ham.n_spinorbitals,) * 2, dtype=complex)
+    twordm = np.zeros((ferm_ham.n_spinorbitals,) * 4, dtype=complex)
+    onerdm_spinsum = np.zeros((ferm_ham.n_spinorbitals//2,) * 2, dtype=complex)
+    twordm_spinsum = np.zeros((ferm_ham.n_spinorbitals//2,) * 4, dtype=complex)
+
+    # Check if the data dict is expectation values or raw frequency data in the {basis: hist} format
+    if isinstance(exp_data, dict) and isinstance(list(exp_data.values())[0], float):
+        exp_vals = {pauli_string_to_of(term): exp_val for term, exp_val in exp_data.items()}
+    elif isinstance(exp_data, dict) and isinstance(list(exp_data.values())[0], dict):
+        exp_vals = {}
+
+    n_qubits = get_qubit_number(mapping, ferm_ham.n_spinorbitals)
+
+    # Go over all terms in fermionic Hamiltonian
+    for term in ferm_ham.terms:
+        length = len(term)
+
+        if not term:
+            continue
+
+        # Fermionic term with a prefactor of 1.0.
+        fermionic_term = FermionOperator(term, 1.0)
+
+        qubit_term = fermion_to_qubit_mapping(fermion_operator=fermionic_term,
+                                            n_spinorbitals=ferm_ham.n_spinorbitals,
+                                            n_electrons=ferm_ham.n_electrons,
+                                            mapping=mapping,
+                                            up_then_down=up_then_down)
+        qubit_term.compress()
```

Comment:
'scbk' requires spin so that should also be defined in ferm_ham

Reply:
Done